sdk/iot/provisioning/inc/az_iot_provisioning_client.h

```diff
@@ -165,15 +186,19 @@ AZ_NODISCARD az_result az_iot_hub_client_sas_password_get(
  * @brief Gets the MQTT topic filter to subscribe to register responses.
  *
  * @param[in] client The #az_iot_provisioning_client to use for this call.
- * @param[in] mqtt_topic_filter An empty #az_span with sufficient capacity to hold the MQTT topic
- *                              filter.
- * @param[out] out_mqtt_topic_filter The output #az_span containing the MQTT topic filter.
+ * @param[out] mqtt_topic_filter A buffer with sufficient capacity to hold the MQTT topic filter.
+ *                               If successful, contains a null-terminated string with the topic 
+ *                               filter that needs to be passed to the MQTT client.
+ * @param[in] mqtt_topic_filter_size The size, in bytes of #mqtt_topic_filter.

```

Comment:
It is a bit odd that `mqtt_topic_filter` is [out] but the `mqtt_topic_filter_size` is [in]. Is the size, the maximum size we can write into the ptr?

Reply:
Yes. I'm now worried it's not clear from the docs. Ideas how to improve?

Reply:
We could try to doc that it is the maximum available size of the buffer that this method can write into similar to `az_span_to_str`:
https://github.com/Azure/azure-sdk-for-c/blob/31b0dd7d8baea135505bac36d5145f6d018c7f56/sdk/core/core/inc/az_span.h#L217-L219

Reply:
I kept it in-line with the span (buffer, size). I'd like to settle on the right names - changing these is time consuming.
Since the pattern I'm using was already approved by the API review board, please re-open the discussion in the next meeting.

Reply:
Parameter name aside, maybe we can just update the doc comment to be more clear:

```c
 * @param[in] mqtt_topic_filter_size The maximum available space, in bytes, within the buffer referred to by \p mqtt_topic_filter.
```